feat: add library search

ninjaasmoke opened the pull request:
### Description:

- **Feature**: Adds a search bar to the Libraries feature, allowing users to filter library items by name as they type in a search query.
- **Usage**: The search input dynamically filters the displayed items, helping users find specific components more quickly, especially within larger libraries.
- **Implementation**:
  - Utilized useMemo for optimized filtering of library items based on the current search query.
  - Included case-insensitive matching for more user-friendly searches.
- **Benefits**:
  - Improves user experience by making it easier to find components within published libraries.
  - Reduces time spent scrolling through extensive libraries, especially for users who frequently reuse specific components.

ninjaasmoke commented:
why is no one looking at PRs?🥲

Mrazator reviewed (commented):
> why is no one looking at PRs?🥲

Hey, just let us know it's ready - i.e. by commenting or assigning one of the maintainers to the review. There are a ton of PRs in a half-baked state and we can't afford to watch all of them.

I think it looks good, though I would try to re-use `QuickSearch` component, there shall be no need for yet another search logic + styles.

ninjaasmoke commented:
hey @Mrazator, thanks for the suggestion. Changed implementation to use existing `QuickSearch` component. Please have a look.

ninjaasmoke commented:
> * The style should match the greyish one in the text search

Would it be okay if I replaced the text search with `QuickSearch` and updated `QuickSearch` to match the greyish style of the current text search? This change would also unify the search UI across these areas:

1. FontPickerList (fonts)
2. UserList (collaborators)

YonatanMimoun commented:
Any progress? I’m just a user of the tool, but I really think this feature would be a huge help!

ninjaasmoke commented:
@Mrazator 
1. moved quick search to top (can be used to filter private lib items when they have names)
2. normalised the search style to match the one in text search (same for the `FontPickerList` and `UserList`

Please have a look at it.

yanir21 commented:
Bumping this PR as a user of the tool. This feature is the only thing I'm waiting for to fully switch from other tools.

jankrloz commented:
Basic feature, please prioritize this!

Mrazator commented:
@ninjaasmoke I am sorry, I missed the mention somehow. 

From functional perspective I think it looks good, IMO it's already useful. Though a bit half-baked, since we are still missing "1) all library items having "names" prop 2) UI for displaying library item names 3) UI/UX for specifying private library item name", as mentioned above. 

@dwelle thoughts?

Mrazator commented on packages/excalidraw/components/LibraryMenuItems.tsx:
```diff
@@ -321,6 +351,11 @@ export default function LibraryMenuItems({
           ) : null}
+          {filteredPublishedItems.length === 0 &&
+            inputRef &&
+            publishedSearchQuery.length && (
```
Note that this might be causing the `0` to be rendered.
<img width="285" alt="Screenshot 2025-02-21 at 23 06 32" src="https://github.com/user-attachments/assets/e2dc384a-c86e-4dc5-9e18-b40cb8d01b9b" />

Fenweldryn commented:
There are currently two search features pending approval. Can we have one of those PRs approved pretty please? 😬

The-Luc commented:
Any new update on this
